### PR TITLE
Bump up the max number of code points for JSON/YAML parser DAT-12657

### PR DESCRIPTION
**Pull request by wwillard7800**
Refactored the creation of the Yaml object in order to pass a LoaderOptions instance which sets the maximum number of code points higher so that larger JSON files can be parsed.

### CODE REVIEW

**Review by filipelautert — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java [hunk 1]
```diff
@@ -11,19 +11,25 @@
...
+import org.yaml.snakeyaml.representer.Representer;
+import org.yaml.snakeyaml.resolver.Resolver;
 
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.util.Map;
 
 public class YamlSnapshotParser extends YamlParser implements SnapshotParser {
 
+    public static final int CODE_POINT_LIMIT = 9 * 1024 * 1024;
```
Comment (StevenMassaro) [inline liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java:27]: Why not just make this MAX_VALUE? I see no reason to limit the size on our end.

File: liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java [hunk 2]
```diff
@@ -11,19 +11,25 @@
...
+
     @SuppressWarnings("java:S2095")
     @Override
     public DatabaseSnapshot parse(String path, ResourceAccessor resourceAccessor) throws LiquibaseParseException {
-        Yaml yaml = new Yaml(new SafeConstructor());
+        Yaml yaml = createYaml();
```
Comment (StevenMassaro) [inline liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java:32]: We create a `Yaml` object in the pro code too, I wonder if we should increase the code point limit there as well.

File: liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java (hunk 1 above)
Comment (wwillard7800) [reply liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java:27]: Agreed.

File: liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java (hunk 2 above)
Comment (wwillard7800) [reply liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java:32]: I'll take a look.

Comment (StevenMassaro) [reply liquibase-core/src/main/java/liquibase/parser/core/yaml/YamlSnapshotParser.java:32]: This may be best done with another story, up to you.

**Review by suryaaki2 — APPROVED** [review]

**Pull request merged**